packages/components/src/mobile/dark-mode/index.native.js

```diff
@@ -47,7 +47,11 @@ export function withTheme( WrappedComponent ) {
 		}
 
 		render() {
-			return <WrappedComponent theme={ this.state.mode } { ...this.props } />;
+			return <WrappedComponent
+				theme={ this.state.mode }
+				useStyle={ ( light, dark ) => useStyle( light, dark, this.state.mode ) }
+				{ ...this.props }
+			/>;
```

Comment:
What would you think about deleting the freestanding `useStyle` function that takes the theme parameter and instead just passing a class function along these lines:
```
return class extends React.Component {
  ...
  render() {
    return <WrappedComponent
      theme={ this.state.mode }
      useStyle={ this.useStyle }
      { ...this.props }
      />;
  }
  
  useStyle( light, dark ) {
    ...
    return this.state.mode === 'dark' ? finalDark : light;
  }
};
```

Just a thought, certainly not a blocker for this PR. 

A simpler version of this change would be to just not export the current `useStyle` function to make it clear that it should only be used via the `withTheme` wrapper component. The only (pretty minor) negative of that approach is that it makes the connection between the `mode` state and the `useStyle` function slightly less obvious because you have to look to where the `useStyle` function is called in the render method instead of just looking at the `useStyle` method itself to see how `this.state.mode` is used.

Reply:
> What would you think about deleting the freestanding useStyle function that takes the theme parameter and instead just passing a class function

Absolutely! Will do that change 👍